From fbbf71f3740939c44dfb5411e2daf3bf0d8e8f41 Mon Sep 17 00:00:00 2001
From: Dmitry Baryshkov <dmitry.baryshkov@linaro.org>
Date: Fri, 10 Jun 2022 21:55:41 +0300
Subject: phy: qcom-qmp: fix msm8996 PCIe PHY support

From: Dmitry Baryshkov <dmitry.baryshkov@linaro.org>

commit fbbf71f3740939c44dfb5411e2daf3bf0d8e8f41 upstream.

Replace init/exit ops with power_on/power_off which should be used for
the PCIe PHYs to fix PHY initialization.

Fixes: f575ac2d64e7 ("phy: qcom-qmp-pcie-msm8996: drop support for non-PCIe PHY types")
Signed-off-by: Dmitry Baryshkov <dmitry.baryshkov@linaro.org>
Link: https://lore.kernel.org/r/20220610185542.3662484-2-dmitry.baryshkov@linaro.org
Signed-off-by: Vinod Koul <vkoul@kernel.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---
 drivers/phy/qualcomm/phy-qcom-qmp-pcie-msm8996.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/drivers/phy/qualcomm/phy-qcom-qmp-pcie-msm8996.c
+++ b/drivers/phy/qualcomm/phy-qcom-qmp-pcie-msm8996.c
@@ -867,8 +867,8 @@ static int phy_pipe_clk_register(struct
 }
 
 static const struct phy_ops qcom_qmp_phy_pcie_msm8996_ops = {
-	.init		= qcom_qmp_phy_pcie_msm8996_enable,
-	.exit		= qcom_qmp_phy_pcie_msm8996_disable,
+	.power_on	= qcom_qmp_phy_pcie_msm8996_enable,
+	.power_off	= qcom_qmp_phy_pcie_msm8996_disable,
 	.set_mode	= qcom_qmp_phy_pcie_msm8996_set_mode,
 	.owner		= THIS_MODULE,
 };
